wazo_confd/plugins/user_blocklist/validator.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from xivo_dao.alchemy.blocklist import BlocklistNumber
+from xivo_dao.helpers import errors
+from xivo_dao.resources.blocklist import dao
+
+from wazo_confd.helpers.validator import ValidationGroup, Validator
+
+
+class UniqueNumberInBlocklist(Validator):
+    def __init__(self, dao, update=False):
+        super().__init__()
+        self.dao = dao
+        self.update = update
+
+    def validate(self, model: BlocklistNumber):
+        existing = self.dao.find_by(
+            number=model.number, blocklist_uuid=model.blocklist_uuid
+        )
+        if existing and (not self.update or existing.uuid != model.uuid):
+            raise errors.resource_exists(
+                'BlocklistNumber',
+                number=model.number,
+                blocklist_uuid=model.blocklist_uuid,
+            )
```

Comment:
This is hard to read, it took me a while to understand the logic. Suggestion:
```suggestion
        if not existing:
            return
        if self.update and existing.uuid == model.uuid:
            return
        raise errors.resource_exists(
                'BlocklistNumber',
                number=model.number,
                blocklist_uuid=model.blocklist_uuid,
            )
```